### PR TITLE
Support SharePlay scrubbing

### PR DESCRIPTION
**Pull request by mukulmishra18**
We want to support SharePlay scrubbing in our player ui.

This PR implement the support for it. The changes required for the support:
- Seek event handling is delayed for 100ms if GroupPlayback API is available
- If UI is still in seeking state after 100ms(if there is no `mouseup` after `mousedown` event), then suspend group playback
- Local playback will be paused during the scrubbing 
- After end of scrubbing, local playback will be resumed and `endSuspension` is called on `groupPlayback`

### CODE REVIEW

**Review by wasp898 — APPROVED** [review]
LGTM.

File: src/ts/components/seekbar.ts [hunk 1]
```diff
@@ -344,8 +350,21 @@ export class SeekBar extends Component<SeekBarConfig> {
...
+    };
 
+    this.onSeek.subscribe((sender) => {
+      // track seeking status so we can catch events from seek preview seeks
+      this.isUserSeeking = true;
+      // Notify UI manager of started seek
+      uimanager.onSeek.dispatch(sender);
+
+      if (isGroupPlaybackAPIAvailable(player) && player.groupPlayback.hasJoined()) {
+        setTimeout(() => handleSeekEvent(), 100);
```
Comment (stonko1994) [inline src/ts/components/seekbar.ts:362]: After discussing the delay handling offline with @hawk23 we have the feeling that this is not really needed. We don't see any drawback if we always start the suspension no matter if seek happened by scrubbing or just by tapping.

File: src/ts/components/seekbar.ts [hunk 2]
```diff
@@ -328,11 +330,15 @@ export class SeekBar extends Component<SeekBarConfig> {
...
-      this.isUserSeeking = true; // track seeking status so we can catch events from seek preview seeks
+    let isGroupPlaybackAPIAvailable = (player: PlayerAPI): player is ExtendedPlayerAPI => {
+      return !!(player as ExtendedPlayerAPI).groupPlayback;
+    };
 
-      // Notify UI manager of started seek
-      uimanager.onSeek.dispatch(sender);
+    let handleSeekEvent = () => {
+      // If group playback is present and user still seeking after 100ms then it might be a scrubbing.
+      if (isGroupPlaybackAPIAvailable(player) && player.groupPlayback.hasJoined() && this.isUserSeeking && !suspension) {
```
Comment (stonko1994) [inline src/ts/components/seekbar.ts:339]: My UI knowledge got a bit rusty over the last years but can't we use the `scrubbing` flag for this potentially? 🤔

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [development]
+
+### Added
+- Support for SharePlay scrubbing
```
Comment (stonko1994) [inline CHANGELOG.md:10]: Nit:

```suggestion
### Added
- Scrubbing during a FaceTime call automatically creates a SharePlay suspension to not synchronize scrubbing to other participants.
```

File: src/ts/components/seekbar.ts (hunk 2 above)
Comment (mukulmishra18) [reply src/ts/components/seekbar.ts:339]: hmm `scrubbing` is set on `mousemove` and that might not be true after the timeout. But as now we always want to suspend we don't need it 😄

Comment (mukulmishra18) [reply src/ts/components/seekbar.ts]: Removed https://github.com/bitmovin/bitmovin-player-ui/pull/533/commits/e99a9f453c23778e6378521bbc64351b278ca211

**Review by stonko1994 — APPROVED** [review]

**Pull request merged**